iOS 12.2.0.GA: backgroundRepeat crashes app

### I have searched and made sure there are no existing issues for the issue I am filing

- [X] I have searched the existing issues

### Description

Setting backgroundRepeat on a view will crash app

```
[ERROR] Application received error: UIGraphicsBeginImageContext() failed to allocate CGBitampContext: size={0, 0}, scale=3.000000, bitmapInfo=0x2002. Use UIGraphicsImageRenderer to avoid this assert.
[ERROR]     0   CoreFoundation                      0x00000001804658a8 __exceptionPreprocess + 172
[ERROR]     1   libobjc.A.dylib                     0x000000018005c09c objc_exception_throw + 56
[ERROR]     2   Foundation                          0x0000000180cf3b90 -[NSMutableDictionary(NSMutableDictionary) classForCoder] + 0
[ERROR]     3   UIKitCore                           0x0000000116b90f68 _UIGraphicsBeginImageContextWithOptions + 564
[ERROR]     4   TitaniumKit                         0x000000010665f060 -[TiUIView renderRepeatedBackground:] + 248
[ERROR]     5   TitaniumKit                         0x000000010665f2c4 -[TiUIView setBackgroundImage_:] + 344
[ERROR]     6   TitaniumKit                         0x0000000106683bd0 DoProxyDelegateReadValuesWithKeysFromProxy + 424
[ERROR]     7   TitaniumKit                         0x000000010666c9ec -[TiViewProxy firePropertyChanges] + 68
[ERROR]     8   TitaniumKit                         0x000000010666b754 -[TiViewProxy view] + 200
[ERROR]     9   TitaniumKit                         0x000000010666b7c4 -[TiViewProxy view] + 312
[ERROR]     10  TitaniumKit                         0x000000010664a78c -[TiWindowProxy openOnUIThread:] + 56
[ERROR]     11  libdispatch.dylib                   0x00000001801424f4 _dispatch_call_block_and_release + 24
[ERROR]     12  libdispatch.dylib                   0x0000000180143d3c _dispatch_client_callout + 16
[ERROR]     13  libdispatch.dylib                   0x0000000180152b24 _dispatch_main_queue_drain + 1272
[ERROR]     14  libdispatch.dylib                   0x000000018015261c _dispatch_main_queue_callback_4CF + 40
[ERROR]     15  CoreFoundation                      0x00000001803c61b4 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 12
[ERROR]     16  CoreFoundation                      0x00000001803c08cc __CFRunLoopRun + 1936
[ERROR]     17  CoreFoundation                      0x00000001803bfd28 CFRunLoopRunSpecific + 572
[ERROR]     18  GraphicsServices                    0x000000018986ebc0 GSEventRunModal + 160
[ERROR]     19  UIKitCore                           0x0000000116fdffdc -[UIApplication _run] + 868
[ERROR]     20  UIKitCore                           0x0000000116fe3c54 UIApplicationMain + 124
[ERROR]     21  listViewWithVideo                   0x00000001021fcbf0 main + 848
[ERROR]     22  dyld                                0x0000000102729558 start_sim + 20
[ERROR]     23  ???                                 0x00000001028b1f28 0x0 + 4337639208
[ERROR]     24  ???                                 0x0a38800000000000 0x0 + 736479276563431424
```

### Expected Behavior

backgroundImage should repeat

### Actual behavior

App crashes

### Reproducible sample

```
const tabGroup = Ti.UI.createTabGroup();

tabGroup.addTab(createTab('Tab 1', 'I am Window 1', 'assets/images/tab1.png'));
tabGroup.addTab(createTab('Tab 2', 'I am Window 2', 'assets/images/tab2.png'));

tabGroup.open();

createTab(title, message, icon) {
    const window = Ti.UI.createWindow({
        backgroundColor: 'backgroundColor',
        title: title
    });

    const view = Ti.UI.createView({
        backgroundImage: 'assets/images/tab1.png',
        backgroundRepeat: true,
        height: 50,
    });

    window.add(view);

    return Ti.UI.createTab({
        title: title,
        icon: icon,
        window: window
    });
}

```

### Steps to reproduce

Run sample

### Platform

iOS

### SDK version you are using

SDK 12.2.0.GA

### Alloy version you are using

_No response_

--- a/iphone/TitaniumKit/TitaniumKit/Sources/API/TiUIView.m
+++ b/iphone/TitaniumKit/TitaniumKit/Sources/API/TiUIView.m
@@ -658,23 +658,29 @@ DEFINE_EXCEPTIONS
   // drawing operations on iOS (and presumably Android). By removing this code and instead blitting the [bgImage CGImage]
   // directly into the graphics context, it tesselates from the lower-left.
 
-  UIGraphicsBeginImageContextWithOptions(bgImage.size, NO, bgImage.scale);
-  CGContextRef imageContext = UIGraphicsGetCurrentContext();
-  CGContextDrawImage(imageContext, CGRectMake(0, 0, bgImage.size.width, bgImage.size.height), [bgImage CGImage]);
-  UIImage *translatedImage = UIGraphicsGetImageFromCurrentImageContext();
-  UIGraphicsEndImageContext();
+  // Creating the translatedImage
+  UIGraphicsImageRendererFormat *format1 = [[UIGraphicsImageRendererFormat alloc] init];
+  format1.scale = bgImage.scale;
 
-  UIGraphicsBeginImageContextWithOptions(self.bounds.size, NO, bgImage.scale);
-  CGContextRef background = UIGraphicsGetCurrentContext();
-  if (background == nil) {
+  UIGraphicsImageRenderer *renderer1 = [[UIGraphicsImageRenderer alloc] initWithSize:bgImage.size format:format1];
+  UIImage *translatedImage = [renderer1 imageWithActions:^(UIGraphicsImageRendererContext *_Nonnull context) {
+    [bgImage drawAtPoint:CGPointZero];
+  }];
+
+  // Creating the renderedBg
+  UIGraphicsImageRendererFormat *format2 = [[UIGraphicsImageRendererFormat alloc] init];
+  format2.scale = bgImage.scale;
+
+  UIGraphicsImageRenderer *renderer2 = [[UIGraphicsImageRenderer alloc] initWithSize:self.bounds.size format:format2];
+  UIImage *renderedBg = [renderer2 imageWithActions:^(UIGraphicsImageRendererContext *_Nonnull context) {
+    UIImage *tiledImage = [UIImage imageWithCGImage:translatedImage.CGImage scale:bgImage.scale orientation:translatedImage.imageOrientation];
+    [tiledImage drawAsPatternInRect:context.format.bounds];
+  }];
+
+  if (renderedBg == nil) {
     //TIMOB-11564. Either width or height of the bounds is zero
-    UIGraphicsEndImageContext();
     return;
   }
-  CGRect imageRect = CGRectMake(0, 0, bgImage.size.width, bgImage.size.height);
-  CGContextDrawTiledImage(background, imageRect, [translatedImage CGImage]);
-  UIImage *renderedBg = UIGraphicsGetImageFromCurrentImageContext();
-  UIGraphicsEndImageContext();
 
   [self backgroundImageLayer].contents = (id)renderedBg.CGImage;
 }

